app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer.go

```diff
@@ -251,11 +252,32 @@ func (l *LoadBalancerSyncer) DeleteLoadBalancer(ing *v1beta1.Ingress) error {
 
 // PrintStatus prints the current status of the load balancer.
 func (l *LoadBalancerSyncer) PrintStatus() (string, error) {
-	sd, err := l.frs.GetLoadBalancerStatus(l.lbName)
-	if err != nil {
-		return "", err
-	}
-	return fmt.Sprintf("Load balancer %s has IPAddress %s and is spread across %d clusters (%s)", l.lbName, sd.IPAddress, len(sd.Clusters), strings.Join(sd.Clusters, ",")), nil
+	// First try to fetch the status from url map.
+	// If that fails, then we fetch it from forwarding rule.
+	// This is because we first used to store the status on forwarding rules and then migrated to url maps.
+	// https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/issues/145 has more details.
+	sd, umErr := l.ums.GetLoadBalancerStatus(l.lbName)
+	if umErr == nil {
+		return formatLoadBalancerStatus(l.lbName, *sd), nil
+	}
+	if ingressutils.IsHTTPErrorCode(umErr, http.StatusNotFound) {
+		return "", fmt.Errorf("Load balancer %s does not exist", l.lbName)
+	}
+	// Try forwarding rule.
+	sd, frErr := l.frs.GetLoadBalancerStatus(l.lbName)
+	if frErr == nil {
+		return formatLoadBalancerStatus(l.lbName, *sd), nil
+	}
+	if ingressutils.IsHTTPErrorCode(frErr, http.StatusNotFound) {
+		return "", fmt.Errorf("Load balancer %s does not exist", l.lbName)
+	}
+	// Failed to get status from both url map and forwarding rule.
+	return "", fmt.Errorf("failed to get status from both url map and forwarding rule. Error in extracting status from url map: %s. Error in extracting status from forwarding rule: %s", umErr, frErr)
```

Comment:
I believe this error is shown to the user, right? Can this be a little more friendlier than printing out the url map and forwarding rule details? May be just say there was an error extracting the status and print all these details in the log?